zio/csupio/vectorreader.go

```diff
@@ -33,24 +31,11 @@ func NewVectorReader(ctx context.Context, zctx *super.Context, r io.Reader, push
 	if !ok {
 		return nil, errors.New("Super Columnar requires a seekable input")
 	}
-	objects, err := readObjects(ra)
-	if err != nil {
-		return nil, err
-	}
-	var metaFilter expr.Evaluator
-	var metaProjection field.Projection
-	if pushdown != nil {
-		metaFilter, metaProjection, err = pushdown.MetaFilter()
-		if err != nil {
-			return nil, err
-		}
-	}
 	return &VectorReader{
 		ctx:           ctx,
 		zctx:          zctx,
 		activeReaders: &atomic.Int64{},
-		nextObject:    &atomic.Int64{},
-		objects:       filterObjects(zctx, metaFilter, objects, metaProjection),
+		stream:        &stream{r: ra},
```

Comment:
You're not setting the pushdown field here and you should right?

Reply:
True!

Reply:
yikes